x-pack/platform/plugins/shared/onechat/server/routes/index.ts

```diff
@@ -10,10 +10,14 @@ import { registerToolsRoutes } from './tools';
 import { registerAgentRoutes } from './agents';
 import { registerChatRoutes } from './chat';
 import { registerConversationRoutes } from './conversations';
+import { registerMCPRoutes } from './mcp';
 
 export const registerRoutes = (dependencies: RouteDependencies) => {
   registerToolsRoutes(dependencies);
   registerAgentRoutes(dependencies);
   registerChatRoutes(dependencies);
   registerConversationRoutes(dependencies);
+  if (dependencies.config.mcpServer.enabled) {
```

Comment:
one thing we need to talk about is feature flag approach. This uses the env config variable approach which is difficult in practice for us to enable the feature for a project. Requires a PR change at the project-controller level to able. Lets get clarity on our strategy here. will open a thread on this.